compiler/compiler.go

```diff
@@ -379,16 +447,21 @@ func compile(n semantic.Node, subst map[uint64]semantic.MonoType, scope Scope) (
 		if err != nil {
 			return nil, err
 		}
-		lt := l.Type()
+		lt := l.Type().Nature()
 		r, err := compile(n.Right, subst, scope)
 		if err != nil {
 			return nil, err
 		}
-		rt := r.Type()
+		rt := r.Type().Nature()
+		if lt == semantic.Invalid {
```

Comment:
This means the left hand side is null?

Reply:
Yes. This is to work around needing to mock out nulls everywhere in the binary lookup.